frontend/src/components/Pages/InspectionPage/ScheduleMissionDialog.tsx

```diff
@@ -0,0 +1,159 @@
+
+    useEffect(() => {
+        const id = setInterval(() => {
+            BackendAPICaller.getEnabledRobots().then((robots) => {
+                setRobotOptions(robots)
+            })
+        }, props.refreshInterval)
+        return () => clearInterval(id)
+    }, [props.refreshInterval])
+
+    let timer: ReturnType<typeof setTimeout>
+
+    const onSelectedRobot = (selectedRobot: Robot) => {
+        if (robotOptions === undefined) return
+
+        setSelectedRobot(selectedRobot)
+    }
+
+    const onScheduleButtonPress = () => {
+        if (selectedRobot === undefined) return
+
+        props.missions.forEach((mission) => BackendAPICaller.scheduleMissionDefinition(mission.id, selectedRobot.id))
+
+        setSelectedRobot(undefined)
+    }
+
+    const closePopover = () => setIsPopoverOpen(false)
+
+    const handleClose = () => {
+        clearTimeout(timer)
+        closePopover()
+    }
+
+    return (
+        <>
+            <Popover
+                anchorEl={anchorRef.current}
+                onClose={handleClose}
+                open={isPopoverOpen && installationCode === ''}
+                placement="top"
+            >
+                <Popover.Content>
+                    <Typography variant="body_short">{TranslateText('Please select installation')}</Typography>
+                </Popover.Content>
+            </Popover>
+
+            <StyledMissionDialog>
+                <StyledDialog open={true}>
+                    <StyledAutoComplete>
+                        <StyledMissionSection>
+                            <Button
+                                onClick={() => {
+                                    props.closeDialog()
+                                }}
+                                variant="outlined"
+                                color="primary"
+                            >
+                                {' '}
+                                {TranslateText('Cancel')}{' '}
+                            </Button>
+                        </StyledMissionSection>
+                    </StyledAutoComplete>
+                </StyledDialog>
+            </StyledMissionDialog>
+
+            <StyledMissionDialog>
+                <Dialog open={true}>
+                    <StyledAutoComplete>
+                        <Typography variant="h5">{TranslateText('This installation has no missions')}</Typography>
+                    </StyledAutoComplete>
+                </Dialog>
+            </StyledMissionDialog>
+
+            <StyledMissionDialog>
+                <StyledDialog open={true}>
+                    <StyledAutoComplete>
+                        <Autocomplete
+                            optionLabel={(r) => r.name + ' (' + r.model.type + ')'}
+                            options={robotOptions.filter(
+                                (r) =>
+                                    r.currentInstallation.toLocaleLowerCase() === installationCode.toLocaleLowerCase()
```

Comment:
Suggest filtering the robots with the installation code on the use effect instead of here so that maybe the autocomplete option load a bit faster

Reply:
I've tried and didn't help much but still think it looks better here than in the autocomplete

useEffect(() => {
        const id = setInterval(() => {
            BackendAPICaller.getEnabledRobots().then((robots) => robots.filter(robots => robots.currentInstallation.toLowerCase() === installationCode.toLowerCase()))
            .then((robots) => {
                setRobotOptions(robots)
            })
        }, props.refreshInterval)
        return () => clearInterval(id)
    }, [props.refreshInterval])